doc/tutorials/ostro-oic-tutorial.md

```diff
@@ -0,0 +1,1030 @@
+features of Soletta, like OIC and HTTP communication, besides GPIO
+interaction.
+Initially, we are going to run the application on desktop [1], but
+later we're going to show how to run it on devices, such as [MinnowBoard
+MAX](http://wiki.minnowboard.org/MinnowBoard_Wiki_Home).
+
+## Step 0
+
+Soletta must be installed on developer's computer. One can use a package
+provide for his or her distro, or [build Soletta](https://github.com/solettaproject/soletta/wiki/How-to-start).
+
+## Step 1
+
+### Flow Based Programming overview
+
+Soletta provides two API [3] for developers. A C API and a flow based
+API. In this tutorial, we're going to show the flow based API.
+The flow based API uses [Flow Based Programming](https://en.wikipedia.org/wiki/Flow-based_programming)
+concepts. It is a higher level API than its C counterpart, and it's
+much easier to use when expressing relationships among application
+components.
+Basically, a flow based application contains 'nodes' which represents
+entities - like sensors or actuators, as well as operations like
+concatenating string. These nodes have input or output 'ports' that
+send or receive 'packets' of information. These ports can be connected
+to express the flow of information among components. For instance:
+
+```
+button(gpio/reader:pin=1) OUT -> IN light(gpio/writer:pin=2)
+```
+
+Here, `button` is a node of node type [4] `gpio/reader` and light is
+another node, of node type `gpio/writer`. A `gpio/reader` node
+has an output port `OUT` - which sends boolean packets - that
+is connected to input port `IN` of the `gpio/writer` - which
+in turn, receives boolean packets as well. So, in this sample,
+we use `button` output to feed `light`. The `pin=N` on those
+nodes is a 'node option'. Node options are used to setup nodes,
+so each node, in this sample, is aware of which GPIO pin they
+represent.
+Note that subsequent uses of a node don't take its setup information
+again, so, if we wanted to debug `button` output to console,
+we would do:
+
+```
+button OUT -> IN _(console)
+```
+
+For more information about Soletta flow programming approach,
+you can see [Soletta README](https://github.com/solettaproject/soletta#flow-based-programming).
+For more information about Soletta available node types,
+see Soletta node type [Cheat Sheet](http://solettaproject.github.io/docs/nodetypes/).
+
+### Writing a **really** simple application
+
+The first step to write an application is to define what it shall do.
+Let's imagine that we want to control a device that has two features:
+'state' and 'power'. One can think of a light, for instance. It's either
+'ON' or 'OFF' - the state, and has different light intensities - the power.
+We can use two accumulators for this task: one from 0 to 1, to store
+'state', and another from 0 to 100 to store 'power'.
+
+```
+power_value(int/accumulator:setup_value=min:0|max:100|step:1,initial_value=50)
+state_value_as_int(int/accumulator:setup_value=min:0|max:1)
+```
```

Comment:
well, this part is kinda cumbersome I've to admit. Maybe start with the simpler part that is the server itself. You can start by defining the server using http, use curl as controller. You can use a tool most people are familar with to query and set the value. Then you elaborate on the server and keep the controller like that. Follow with an http controller and then add oic, with oic controller. The learning curve won't be as steep.